Fix action calculation

MarcelBruckner opened the pull request.

MarcelBruckner committed:
Low high observation space calculation

MarcelBruckner committed:
High low calc in normalize path

MarcelBruckner committed:
Added NearestAgentObserver

MarcelBruckner committed:
Removed the high low env stuff

MarcelBruckner committed:
Added bounds to denormalize in load function

MarcelBruckner committed:
Added observer to test env

MarcelBruckner committed:
Added sanity test for action calculation on sac expert trajectories

MarcelBruckner committed:
Action calculation test green

MarcelBruckner committed:
Readded sac expert trajectory test data

marcelbrucker commented:
Looks good to me.

Only too things:
1. Why this change?
![Bildschirmfoto 2020-08-02 um 12 30 11](https://user-images.githubusercontent.com/63588974/89121078-0a4d4e00-d4bc-11ea-90c2-f032716bf0a7.jpg)
2. Some length assertion in load_expert_trajectories_tests.py fails in CI. Is this valid or a problem?

marvinlue committed:
Added unnormalization checking with methods to work in joblib files
Modified the load expert trajectory function to work with unnomalized and normalized input from sac agent
Added the sac env option in gail params and also in the gail script

MarcelBruckner committed:
Merge branch 'master' into marcel_bruckner/fix-action-calculation-on-unnormalized-state

MarcelBruckner committed:
SAC uses NearestObserver (Cpp) for equal usage in bp and generate, fixed
tests

MarcelBruckner committed:
Merge branch 'marcel_bruckner/CI-apt-dependencies' into marcel_bruckner/fix-action-calculation-on-unnormalized-state

MarcelBruckner committed:
Merge branch 'marcel_bruckner/CI-apt-dependencies' into marcel_bruckner/fix-action-calculation-on-unnormalized-state

MarcelBruckner committed:
First successful training on SAC unnormalized observations

MarcelBruckner commented on bark_ml/library_wrappers/lib_tf_agents/runners/sac_runner.py:
```diff
@@ -60,24 +81,35 @@ def GenerateExpertTrajectories(self, num_trajectories: int = 1000, render: bool
         if render:
           self._environment.render()
 
       if info and info['goal_reached']:
```
I added the info check to filter the expert trajectories that end with goal reached. Only if the environment gives a valid last step (The workaround for the index error) and the goal was reached, we add the trajectory to the set of expert trajectories.
The change with `# if True` was just for debugging, so I didn't have to wait for a converged agent :)

MarcelBruckner commented:
Also the CI is fixed in the meantime :+1:

MarcelBruckner committed:
Format and num agents to 3

MarcelBruckner committed:
Remove merge flag from generate script, num scenarios for visualization

MarcelBruckner committed:
Fixed simulation based tests after remove of merge flag

MarcelBruckner commented:
I think we can merge this now. What is your opinion @marcelbrucker ?
If you approve it I will do the merge later today.

MarcelBruckner committed:
Fixed param server for visualize

MarcelBruckner committed:
Removed unnormalization_check

MarcelBruckner committed:
Added generate to normalize test

marcelbrucker commented on examples/example_params/gail_params.json:
```diff
@@ -9,7 +9,8 @@
             "GPUUse": 0,
-            "NormalizeFeatures": true
+            "NormalizeFeatures": true,
+            "SAC":true
```
I believe this SAC parameter is a remnant of @marvinlue and can be deleted.

MarcelBruckner replied:
Removed it :+1:

marcelbrucker reviewed (commented):
Apart from this unnecessary SAC parameter in gail_params.json I approve this pull request.

MarcelBruckner committed:
Removed SAC true artifact

MarcelBruckner committed:
Rename of merging_not_normalized to merging in large data store
